Initial version of installation guide (#76)
DataDog/dd-trace-php: docs/getting_started.md

# Datadog PHP integration

Datadog's PHP APM integration provides the ability to trace critical parts of your PHP applications by automatically instrumenting code and submitting data to Datadog’s APM service using the OpenTracing API.

To provide a great out-of-the-box experience that automatically instruments common libraries without requiring intrusive code changes or wrapping every object, we provide a PHP extension that allows introspecting any PHP function or method.

## Datadog PHP extension installation

Datadog’s PHP extension (`ddtrace`) allows introspection of arbitrary PHP code.

At this moment it is only distributed in source code form, and requires manual compilation.

### Compiling and installing the extension manually

```bash
mkdir dd-trace
cd dd-trace

> curl -L https://github.com/DataDog/dd-trace-php/archive/v0.2.2.tar.gz | tar x --strip-components=1

Reviewer: @pawelchcki and @chuck will also update this to `0.2.3` ack?

Reviewer 2: v0.2.3 ;)

Reviewer: ```suggestion
curl -L https://github.com/DataDog/dd-trace-php/archive/v0.2.3.tar.gz | tar x --strip-components=1
```

## Enabling automatic tracing

Once the C extension is installed, we need to install the PHP package that provides the actual integrations and framework for sending traces to Datadog.

### Install `datadog/dd-trace` package

```bash

composer config minimum-stability beta # required to install opentracing 1.0.0-beta5
composer require opentracing/opentracing
composer require datadog/dd-trace
```

#### Alternative: Install `datadog/dd-trace` package without changing `minimum-stability`

```bash
composer require datadog/dd-trace # first add dd-trace require
# then manually add following entry to your `composer.json` ”require” entry
#  "opentracing/opentracing": "@dev"

#  Example end result should look like:
#  "require": {
#   "datadog/dd-trace": "^0.2.2",
#    "opentracing/opentracing": "@dev"
#  }
# Next run:
composer update
```

### Enabling tracing

#### Laravel integration

To enable Laravel integration we need to configure a new Provider in `config/app.php`

```php

>    'providers' => array_merge(include(base_path('modules/system/providers.php')), [

Reviewer: I feel like we should avoid this `array_merge` detail, which may deviate attention from the line below. Even if you create a new Laravel app the providers definition is a plain array and I think that this is what most developers expect.

```suggestion
   'providers' => [
```

# .....
     'DDTrace\Integrations\LaravelProvider',
```

Now your Laravel application should start sending traces to the Datadog agent running on localhost (in default configuration). The Datadog agent must have APM enabled; see https://docs.datadoghq.com/tracing/setup/ for instructions on installing and configuring the agent.

#### Symfony integration

For Symfony applications, add the bundle in app/AppKernel.php:

```php

>     $bundles = [

Reviewer: ```suggestion
    return [
```

> For Symfony applications, add the bundle in app/AppKernel.php:

Reviewer: Looking at Symfony [docs for bundles](https://symfony.com/doc/current/bundles.html) seems like the file they are suggesting is the  `config/bundles.yml` (and this is where I used to config my bundles actually). Unless is there a specific reason (in which case we should mention it here), shouldn't we propose that approach. I mean, it still works, but maybe what we showing is a less common way to register bundles.

Author: @chuck ^^ ?

Reviewer: 👍 give me a few minutes to setup an app locally.

Reviewer: ```suggestion
For Symfony applications, add the bundle in `config/bundles.php`:
```

        // ...

>         new DDTrace\Integrations\SymfonyBundle(),

Reviewer: This comment is a sort of continuation of the previous one. In that case it would be the class reference and not an instance

Reviewer: ```suggestion
        DDTrace\Integrations\SymfonyBundle::class => ['all' => true],
```

        // ...

>         ];

Reviewer: ```suggestion
    ];
```

```

#### Adding tracing to a function or method

To add spans for another function or method, you can use `dd_trace` to open a span before the code executes, close it when it’s done, set additional tags or errors on the span, and for more advanced usage, modify either the arguments or the return value.

> Here’s an example of tracing the `CustomDriver` class’s `doWork` method, including noting and re-throwing exceptions:

Reviewer: `noting`?

Author: "Note-ing" it threw me off as well, when first read it :)

Reviewer: ```suggestion
Here’s an example of tracing the `CustomDriver` class’s `doWork` method, including reporting any exceptions as errors on the span and re-throwing them:
```
